libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/ServerPickerActivity.java

```diff
@@ -169,10 +171,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
      * @param v View.
      */
     public void setPositiveReturnValue(View v) {
-        setResult(Activity.RESULT_OK, null);
-        final Intent changeServerIntent = new Intent(CHANGE_SERVER_INTENT);
-        sendBroadcast(changeServerIntent);
-        finish();
+        (new AuthConfigTask()).execute();
```

Comment:
This is kicked off when the `Apply` button is clicked on the `Change Server` screen. This indicates a server switch and we need to check if browser-based login is enabled for the new server.